test/actions/admin.test.js

```diff
@@ -858,6 +858,37 @@ describe('Actions.Admin', () => {
         assert.ok(teamAnalytics[TestHelper.basicTeam.id][Stats.USERS_WITH_POSTS_PER_DAY]);
     });
 
+    it('overwritePlugin', async () => {
+        if (TestHelper.isLiveServer()) {
+            console.log('Skipping mock-only test');
+            return;
+        }
+
+        const data1 = fs.createReadStream('test/assets/images/test.png');
+        const data2 = fs.createReadStream('test/assets/images/test.png');
+        const testPlugin = {id: 'testplugin', webapp: {bundle_path: '/static/somebundle.js'}};
+
+        nock(Client4.getBaseRoute()).
+            post('/plugins').
+            reply(200, testPlugin);
+        await Actions.uploadPlugin(data1, false)(store.dispatch, store.getState);
+
+        nock(Client4.getBaseRoute()).
+            post('/plugins').
```

Comment:
While the URL does indeed stay the same, I think the second parameter to `post` accepts the body of the request, and can be used to verify the `force` flag as such. It might be tricky to build the body given the stream, but you can also pass a function to verify, e.g. as per example on https://www.npmjs.com/package/nock#specifying-request-body:
```js
nock('http://www.example.com')
  .post('/login', function(body) {
    return body.username && body.password;
  })
  .reply(200, { id: '123ABC' });
```

This might make the test a little more "useful" (test utility is marginal whenever mocks are involved, but /shrug).

Reply:
@levb, I saw you re-requested a review, but I'm not seeing any changes here: do you think it's worth adding the extra validation to the mock?

Reply:
@lieut-data: my apologies, the changes didn't get added and I missed it. Should be there now. Thanks for the suggestion!